v2/bundle/x509bundle/bundle.go

```diff
@@ -1,65 +1,175 @@
 package x509bundle
 
 import (
+	"bytes"
 	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
 	"io"
+	"io/ioutil"
+	"sync"
 
 	"github.com/spiffe/go-spiffe/v2/spiffeid"
 )
 
+const certType string = "CERTIFICATE"
+
 // Bundle is a collection of trusted public key material for a trust domain.
 type Bundle struct {
+	trustDomain spiffeid.TrustDomain
+	roots       []*x509.Certificate
+	rootsMtx    sync.RWMutex
```

Comment:
nit: group fields protected by the mutex (with the mutex at the top)

```suggestion
	trustDomain spiffeid.TrustDomain
	
	rootsMtx    sync.RWMutex
	roots       []*x509.Certificate

```